spec/requests/post_type_controller_spec.rb

```diff
@@ -0,0 +1,147 @@
+      end
+
+      it 'returns forbidden for accessing post types for other group' do
+        group = create(:group)
+        user = group.leader.user
+        other_group = create(:group)
+        login_as(user)
+        get "/groups/#{other_group.id}/post_types"
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+  end
+
+  describe '#create' do
+    context 'when the user is non-leader' do
+      it 'returns forbidden' do
+        user  = create(:user)
+        group = create(:group)
+        login_as(user)
+        post "/groups/#{group.id}/post_types", post_type: { name: 'fontos' }
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context 'when the user is the group leader' do
+      let(:group) { create(:group) }
+      let(:user) { group.leader.user }
+      before(:each) { login_as(user) }
+
+      it 'redirects back' do
+        post "/groups/#{group.id}/post_types", post_type: { name: 'fontos' }
+        expect(response).to redirect_to(group_path(group))
+      end
+
+      it 'creates a new post type' do
+        post_type_count = PostType.count
+        post "/groups/#{group.id}/post_types", post_type: { name: 'nagyon fontos' }
+        expect(PostType.count).to equal(post_type_count + 1)
+      end
+
+      it 'new post type with empty name redirects back with error message' do
+        post "/groups/#{group.id}/post_types", post_type: { name: '' }
+        expect(response).to redirect_to(group_path(group))
+        expect(flash[:alert]).to be_present
+      end
+
+      it 'returns forbidden for creating post type for other group' do
+        other_group = create(:group)
+        post "/groups/#{other_group.id}/post_types", post_type: { name: 'fontos' }
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+  end
+
+  describe '#destroy' do
+    context 'when the user is non-leader' do
+      it 'returns forbidden' do
+        group      = create(:group)
+        post_type  = create(:post_type, group_id: group.id)
+        other_user = create(:user)
+        login_as(other_user)
+
+        delete "/groups/#{group.id}/post_types/#{post_type.id}"
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context 'when the user is the group leader' do
+      context 'when the post type is not in common types' do
+        it 'actually deletes the post type' do
+          group      = create(:group)
+          post_type  = create(:post_type, group_id: group.id)
+          user       = group.leader.user
+          login_as(user)
+
+          delete "/groups/#{group.id}/post_types/#{post_type.id}"
+
+          expect(PostType.where(id: post_type.id)).not_to exist
+        end
+      end
+
+      context 'when the post type is in common types' do
+        it 'does not delete the post type' do
+          group      = create(:group)
+          user       = group.leader.user
+          leader_post_id = Membership::LEADER_POST_ID
```

Comment:
Regarding the code style, please align the equal signs (using one space after the longest expression) OR use on space after everywhere, this mix looks bad. (I prefer align them, but it's on you)

Also please take a look at the whole file.